src/main/java/software/amazon/awssdk/crt/http/HttpClientConnection.java

```diff
@@ -44,7 +44,19 @@ public HttpStream makeRequest(HttpRequestBase request, HttpStreamResponseHandler
         if (isNull()) {
             throw new IllegalStateException("HttpClientConnection has been closed, can't make requests on it.");
         }
-        HttpStream stream = httpClientConnectionMakeRequest(getNativeHandle(),
+        HttpStreamBase stream = httpClientConnectionMakeRequest(getNativeHandle(),
+                request.marshalForJni(),
+                request.getBodyStream(),
+                new HttpStreamResponseHandlerNativeAdapter(streamHandler));
+
+        return (HttpStream)stream;
+    }
+
+    public HttpStreamBase makeRequest(HttpRequestBase request, HttpStreamBaseResponseHandler streamHandler) throws CrtRuntimeException {
```

Comment:
have the other makeRequest() function be like
`makeRequest(HttpRequest request, HttpStreamResponseHandler streamHandler)`